JoonasHeinonen/escape-from-zordoom-prison
# Feature/expand test level

### [Expand the test level / first technical boss](https://trello.com/c/Orb889UM/77-expand-the-test-level)

https://github.com/JoonasHeinonen/escape-from-zordoom-prison/assets/33086486/19b7eddd-7fc3-403e-8dca-20c761e62ac1

**Changes:**
 - Created GUI for bossfights and logic.
 - Implemented weakspot logic for boss fights.
 - Imported graphical elements for the first bossfight. (Girdeux, but we shall rename him later to something else)

**How to test:**
 - See that the code is well documented and that the inheritance is good from `enemy_base.gd` to `boss_base.gd`.
 - Test the boss fight, see that the backpack takes the hits in 3 seconds and it makes damage.
 - See that the projectiles wont deal any damage from any other part of Girdeux, e.g. shooting him at his face.
 - See that the body works properly.

**Sources cited:**
 - [How to Create UPDATING Health Bars in Godot 4](https://www.youtube.com/watch?v=vkHT5rziNMk&ab_channel=DevWorm)
